zone/command.cpp

```diff
@@ -14106,6 +14107,134 @@ void command_network(Client *c, const Seperator *sep)
 	}
 }
 
+void command_viewzoneloot(Client *c, const Seperator *sep)
+{
+	std::map<uint32,ItemList> zone_loot_list;
+	auto npc_list = entity_list.GetNPCList();
+	uint32 loot_amount = 0, loot_id = 1, search_item_id = 0;
+	if (sep->argnum == 1 && sep->IsNumber(1)) {
+		search_item_id = atoi(sep->arg[1]);
+	} else if (sep->argnum == 1 && !sep->IsNumber(1)) {
+		c->Message(
+			Chat::Yellow,
+			"Usage: #viewzoneloot [item id]"
+		);
+		return;
+	}
+	for (auto npc_entity : npc_list) {
+		auto current_npc_item_list = npc_entity.second->GetItemList();
+		zone_loot_list.insert({ npc_entity.second->GetID(), current_npc_item_list });
+	}
+	for (auto loot_item : zone_loot_list) {
+		uint32 current_entity_id = loot_item.first;
+		auto current_item_list = loot_item.second;
+		auto current_npc = (
+			entity_list.GetNPCByID(current_entity_id) ?
+			entity_list.GetNPCByID(current_entity_id) :
+			nullptr
+		);
+		std::string npc_link;
+		if (current_npc) {
+			std::string npc_name = current_npc->GetCleanName();
+			uint32 instance_id = zone->GetInstanceID();
+			uint32 zone_id = zone->GetZoneID();
+			float npc_x = current_npc->GetX();
+			float npc_y = current_npc->GetY();
+			float npc_z = current_npc->GetZ();
+			std::string command_link = EQ::SayLinkEngine::GenerateQuestSaylink(
+				fmt::format(
+					"#zone {} {} {} {}",
+					zone_id,
+					npc_x,
+					npc_y,
+					npc_z
+				),
+				false,
+				"Goto"
+			);
+			if (instance_id != 0) {
```

Comment:
The only thing different between this `command_link` block and the one before it is whether or not you're in an instance. Could be combined into just one declaration by ternary operating on the `instance_id` for both the command and the second param

eg

```cpp
std::string command_link = EQ::SayLinkEngine::GenerateQuestSaylink(
	fmt::format(
		"#{} {} {} {} {}",
		(instance_id != 0 ? "zoneinstance" : "zone"),
		(instance_id != 0 ? instance_id : zone_id),
		current_npc->GetX(),
		current_npc->GetY(),
		current_npc->GetZ()
	),
	false,
	"Goto"
);
```

Then, because you're just using one declaration, the `float` `npc_x`, `npc_y` etc no longer becomes necessary because you can just use `current_npc->GetX/Y/Z` etc